website/content/specification/_index.md

```diff
@@ -1,50 +1,1320 @@
+{{</callout>}}
+
+### `<prop>`
+
+The optional `<prop>` element provides a structure for declaring arbitrary properties, which consist of a `@namespace`, `@name`, and `@value`.
+
+This data element uses the same syntax as the [`<prop>` allowed on a definition](#prop). When used on an instance, the set of properties MUST apply only to the instance.
+
+Properties declared on the definition MAY be inherited by the instance. Metaschema does not define any general rules for how to handle overlapping and conflicting properties. How to handle these cases SHOULD be defined and documented for each property.
+
+{{<callout>}}
+A property is useful for annotating an instance with additional information that might describe, in a structured way, the semantics, use, nature, or other significant information related to the instance. In many cases, a property might be used to tailor generated documentation or to support an experimental, non-standardized feature in Metaschema.
+{{</callout>}}
+
+### Naming and `<use-name>`
+
+Similar to the [`<use-name>`](#naming-and-use-name) allowed on the referenced definition, the optional `<use-name>` on a instance changes the *effective name* to use for the information element in a compositional data model.
+
+The `<use-name>` element is optional and MAY only occur once.
+
+By default the *effective name* of the information element in a data model is taken from the [*effective name* of the definition](#naming-and-use-name). The `<use-name>` value overrides this behavior for the instance.
+
+{{<callout>}}
+Use of a `<use-name>` frees the module maintainer allowing them to use a sensible *effective name* for the instance in a data model.
+{{</callout>}}
+
+The first matching condition determines the *effective name* for the definition:
+
+1. A `<use-name>` is provided on the instance. The *effective name* is the value of the `<use-name>` element on the instance.
+1. A `<use-name>` is provided on the definition. The *effective name* is the value of the `<use-name>` element on the definition.
+1. No `<use-name>` is provided on the definition. The *effective name* is the value of the `@name` attribute on the definition.
+
+For example:
+
+```xml
+<define-flag name="flag-a">
+  <use-name>flag-b</use-name>
+</define-flag>
+<define-field name="field">
+  <flag ref="flag-a">
+    <use-name>flag-c</use-name>
+  </flag>
+</define-field>
+```
+
+In the example above, the *effective name* of the definition is `flag-c`. If the `<use-name>` was omitted on the instance, the *effective name* would be `flag-b`. If the `<use-name>` was also omitted on the definition, the *effective name* would be `flag-a`.
+
+The following content is valid to the model above.
+
+{{< tabs JSON YAML XML >}}
+{{% tab %}}
+```json
+{
+  "field": {
+    "flag-c": "value"
+  }
+}
+```
+{{% /tab %}}
+{{% tab %}}
+```yaml
+---
+field:
+  flag-c: "value"
+```
+{{% /tab %}}
+{{% tab %}}
+```xml
+<field flag-c="value"/>
+```
+{{% /tab %}}
+{{% /tabs %}}
+
```

Comment:
Do we want to say anything (or is this covered) about not permitting clashing effective names? I.e. siblings (defined together in a model) and their groups must all have distinct names.